api/server/router/local/container.go

```diff
@@ -104,11 +105,29 @@ func (s *router) getContainersLogs(ctx context.Context, w http.ResponseWriter, r
 
 	var since time.Time
 	if r.Form.Get("since") != "" {
-		s, err := strconv.ParseInt(r.Form.Get("since"), 10, 64)
-		if err != nil {
-			return err
+		if strings.Contains(r.Form.Get("since"), ".") {
+			// the fractional portion of the since argument string is in nanoseconds
+			sa := strings.Split(r.Form.Get("since"), ".")
+
+			s, err := strconv.ParseInt(sa[0], 10, 64)
+			if err != nil {
+				return err
+			}
+			n, err := strconv.ParseInt(sa[1], 10, 64)
+			if err != nil {
+				return err
+			}
+			// should already be in nanoseconds but just in case convert n to nanoseonds
+			n *= int64(math.Pow(float64(10), float64(9-len(sa[1]))))
+
+			since = time.Unix(s, n)
+		} else {
+			s, err := strconv.ParseInt(r.Form.Get("since"), 10, 64)
+			if err != nil {
+				return err
+			}
+			since = time.Unix(s, 0)
```

Comment:
Can dedup this a little by doing something like:

```
sa := strings.SplitN(r.Form.Get("since"), 2)
s, err := strconv.ParseInt(sa[0], 10, 64)
if err != nil {
 return err
}
var n int64
if len(sa) == 2 {
 n, err := strconv.ParseInt(sa[1], 10, 64)
 if err != nil {
  return err
 }
}
since = time.Unix(s, n)
```